### PR TITLE
Convert objective if it's not of the requested type

### PR DESCRIPTION
**Pull request by blegat**
It caused this issue https://github.com/jump-dev/DiffOpt.jl/issues/117

### CODE REVIEW

File: src/MOI_wrapper.jl [hunk 1]
```diff
@@ -562,7 +562,7 @@ function MOI.get(
     return MOI.GreaterThan{Float64}(model.variable_info[c.value].lower_bound)
 end
 
-function MOI.get(model::Optimizer, ::MOI.ObjectiveFunction)
+function MOI.get(model::Optimizer, ::MOI.ObjectiveFunction{T})::T where {T}
     return model.objective
```
Comment (odow) [inline src/MOI_wrapper.jl:566]:
```suggestion
    return convert(T, model.objective)
```

The annotation alone is pretty sneaky (I missed in the first read).

Comment (blegat) [reply src/MOI_wrapper.jl:566]: I just copy-pasted the definition from `MOI/src/Utilities/model.jl`.

Comment (odow) [reply src/MOI_wrapper.jl]: Then we should probably also change `model.jl` haha.

Comment (blegat) [reply src/MOI_wrapper.jl]: Agreed, this seems to aspect of the definition seems to be overlooked by many users so making it more visible is a good thing

**Pull request merged**